### DISCUSSION TITLE
Built-in CSS support is disabled with no custom CSS configuration

### DISCUSSION THREAD

**Discussion by HyunSeob** [vercel/next.js#34387] [Help]
### What version of Next.js are you using?

10.2.3

### What version of Node.js are you using?

16.13.1

### What browser are you using?

Firefox

### What operating system are you using?

macOS

### How are you deploying your application?

next export

### Describe the Bug

I have custom webpack rules have `test` and `include` both.
If there is a rule have `test` and `include`, [Webpack loads only files match `test` and `include` both](https://webpack.js.org/configuration/module/#rule-conditions).
But, I saw [code](https://github.com/vercel/next.js/blob/canary/packages/next/build/webpack-config.ts#L1900) that Next.js disable built-in CSS support, when either `test` **or** `include` matches.
So there can be a case disabling built-in CSS support without custom CSS configuration.

### Expected Behavior

Built-in CSS support is not disabled.

### To Reproduce

Add custom webpack configuration rule that have `test` and `include` both:

```js
{
  test: /\.js$/,
  include: () => false,
  // ...
}
```

and then execute `next dev` in your terminal, you can see warning built-in CSS support is disabled.

**Comment by bigsaigon333** [comment]
Hi @HyunSeob!

I couldn't reproduce the bug with the example you mentioned. 

In `canMatchCss` function, when a rule is an instance of RegExp, the function makes the rule to test filenames which ends with '.css', '.scss', '.sass', 'less', 'styl'.

So given that test property is` /\.js$/`, `canMathCss(test)` function never returns true.  

Of course `includes` property is always returning false in the example.

```tsx
    const fileNames = [
      '/tmp/NEXTJS_CSS_DETECTION_FILE.css',
      '/tmp/NEXTJS_CSS_DETECTION_FILE.scss',
      '/tmp/NEXTJS_CSS_DETECTION_FILE.sass',
      '/tmp/NEXTJS_CSS_DETECTION_FILE.less',
      '/tmp/NEXTJS_CSS_DETECTION_FILE.styl',
    ]

    if (rule instanceof RegExp && fileNames.some((input) => rule.test(input))) {
      return true
    }
```

Am I missing something? Please let me know.

I used next@12.1.0. You said your version is 10.2.3, but the code you referred is the latest, so I didn't put an importance on  a version .

**Reply by HyunSeob** [reply]
Hi, @bigsaigon333 

I'm not sure that you understand my point, so I made a [reproducible example](https://github.com/HyunSeob/next-js-built-in-css-support). I hope this example help you.
